drivers/gpio/gpio_sifive.c

```diff
@@ -251,58 +250,37 @@ static int gpio_sifive_pin_interrupt_configure(struct device *dev,
 	volatile struct gpio_sifive_t *gpio = DEV_GPIO(dev);
 	const struct gpio_sifive_config *cfg = DEV_GPIO_CFG(dev);
 
+	gpio->rise_ie &= ~BIT(pin);
+	gpio->fall_ie &= ~BIT(pin);
+	gpio->high_ie &= ~BIT(pin);
+	gpio->low_ie  &= ~BIT(pin);
+
 	switch (mode) {
 	case GPIO_INT_MODE_DISABLED:
-		gpio->rise_ie &= ~BIT(pin);
-		gpio->fall_ie &= ~BIT(pin);
-		gpio->high_ie &= ~BIT(pin);
-		gpio->low_ie  &= ~BIT(pin);
 		irq_disable(gpio_sifive_pin_irq(cfg->gpio_irq_base, pin));
 		break;
 	case GPIO_INT_MODE_LEVEL:
-		/* TODO: The interrupt functionality of this driver is incomplete,
-		 * but for the sake of not slowing down the GPIO API refactor,
-		 * I'm just returning -ENOTSUP until we can track down the issue.
-		 */
-		return -ENOTSUP;
-
-		gpio->rise_ie &= ~BIT(pin);
-		gpio->fall_ie &= ~BIT(pin);
-
+		/* Board supports both levels, but Zephyr does not. */
 		if (trig == GPIO_INT_TRIG_HIGH) {
-			gpio->high_ip  = BIT(pin);
+			gpio->high_ip = BIT(pin);
 			gpio->high_ie |= BIT(pin);
-			gpio->low_ie  &= ~BIT(pin);
-		} else if (trig == GPIO_INT_TRIG_LOW) {
-			gpio->high_ie &= ~BIT(pin);
-			gpio->low_ip   = BIT(pin);
+		} else {
+			__ASSERT_NO_MSG(trig == GPIO_INT_TRIG_LOW);
+			gpio->low_ip = BIT(pin);
 			gpio->low_ie  |= BIT(pin);
 		}
 		irq_enable(gpio_sifive_pin_irq(cfg->gpio_irq_base, pin));
 		break;
 	case GPIO_INT_MODE_EDGE:
-		/* TODO: The interrupt functionality of this driver is incomplete,
-		 * but for the sake of not slowing down the GPIO API refactor,
-		 * I'm just returning -ENOTSUP until we can track down the issue.
-		 */
-		return -ENOTSUP;
-
-		gpio->high_ie &= ~BIT(pin);
-		gpio->low_ie  &= ~BIT(pin);
+		__ASSERT_NO_MSG(GPIO_INT_TRIG_BOTH ==
+				(GPIO_INT_LOW_0 | GPIO_INT_HIGH_1));
 
-		/* Rising Edge, Falling Edge or Double Edge ? */
-		if (trig == GPIO_INT_TRIG_HIGH) {
-			gpio->rise_ip  = BIT(pin);
+		if ((trig & GPIO_INT_HIGH_1) != 0) {
+			gpio->rise_ip = BIT(pin);
 			gpio->rise_ie |= BIT(pin);
-			gpio->fall_ie &= ~BIT(pin);
-		} else if (trig == GPIO_INT_TRIG_LOW) {
-			gpio->rise_ie &= ~BIT(pin);
-			gpio->fall_ip  = BIT(pin);
-			gpio->fall_ie |= BIT(pin);
-		} else {
-			gpio->rise_ip  = BIT(pin);
-			gpio->rise_ie |= BIT(pin);
-			gpio->fall_ip  = BIT(pin);
+		}
+		if ((trig & GPIO_INT_LOW_0) != 0) {
+			gpio->rise_ip = BIT(pin);
```

Comment:
Should be `fall_ip`.

Reply:
Thanks.  Doesn't fix the failures, but clearly wrong.  #23888